jardin/database/base_client.py

```diff
@@ -71,8 +76,10 @@ def execute(self, *query, write=False, **kwargs):
         cursor = None
         try:
             if self._conn is None:
-                self._conn = self.connect_impl()
-            cursor = self.execute_impl(self._conn, *query)
+                with Instrumenter("connection_initiated", tags=self.tags()):
+                    self._conn = self.connect_impl()
+            with Instrumenter("query", tags=self.tags({"query": query})):
```

Comment:
Isn't the query text going to be very high cardinality? Or does that not matter for APM?

Reply:
This will up to the subscriber to handle as they deem fit, for APM, we may want to add the query as a tag for the span, for metrics we probably won't.